core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java

```diff
@@ -126,8 +131,19 @@ public TabletsMetadata build() {
           "scanTable() cannot be used in conjunction with forLevel(), forTable() or forTablet() %s %s",
           level, table);
       if (level == DataLevel.ROOT) {
-        ClientContext ctx = ((ClientContext) _client);
-        return new TabletsMetadata(getRootMetadata(ctx));
+        final ClientContext ctx = ((ClientContext) _client);
+        final RootTabletMetadata rtm = RootTabletMetadata.read(ctx);
+        final SortedMapIterator iter = new SortedMapIterator(rtm.toKeyValues());
+        if (!tabletMetadataFilters.isEmpty()) {
+          for (var filter : tabletMetadataFilters) {
+            if (!filter.acceptRow(iter)) {
+              LOG.trace("Not returning root metadata as it does not pass filter: {}",
+                  filter.getClass().getSimpleName());
+              return new TabletsMetadata((AutoCloseable) null, Set.of());
+            }
+          }
+        }
+        return new TabletsMetadata(rtm.toTabletMetadata());
```

Comment:
I'm not sure if this is working as expected
This only returns either the entire root tablet metadata or nothing. Shouldn't there be a case with some of the data filtered out?

Reply:
This could also use some testing if we have a TabletsMetadata test

Reply:
Thinking this looks good then, but yeah tests would be good.

Reply:
I'll look into this as well